### DISCUSSION TITLE
`include_columns` doesn't seem to adhere to SQLAlchemy’s lower-case-means-insensitive convention

### DISCUSSION THREAD

**Discussion by HYBRID-BEING** [sqlalchemy/sqlalchemy#10256] [Usage Questions]
I was working with third-party MSSQL database and noticed that while tables could be reflected case-insensitively, `include_columns` requires user to provide column names with correct capitalization (and this database has some tables with both lowercase- and uppercase-named columns). The database itself is not case-sensetive, i'm able to select columns case-insensitively using Management Studio.

I believe, related code is here:
https://github.com/sqlalchemy/sqlalchemy/blob/d4dcb2e2176d33d9e7bbee7fe73d34290dbb503e/lib/sqlalchemy/databases/mssql.py#L599-L600 This, of course, means that name of the reflected column is checked for membership against the `include_columns` list as-is.

It's about MySQL, but #1206 is somewhat related. Maybe the lower-case-means-insensitive convention is more of a recent thing, but i think it would make sense to also follow it in the case of `include_columns`, perhaps regardless of dialect.

**Comment by zzzeek** [comment]
confirmed SQL Server columns are case insensitive and you can't have two same-named columns with different cases so we can accept PRs with round trip SQL Server tests for `name.lower() in include_columns` where `include_columns` has been converted to `[n.lower() for n in include_columns]`.

**Reply by HYBRID-BEING** [reply]
Actually, I might be mistaken, but, i think, it's possible to have case sensetivity in SQL Server by configuring collation. Or does it only affect queries, and having two same-named columns with different cases not possible anyway?

**Reply by zzzeek** [reply]
the question is, is it possible to have a table like this:

```
CREATE TABLE sql_server_table (
	[case_Sens] VARCHAR(50) NULL, 
	[Case_Sens] VARCHAR(50) NULL
)

```

right now that raises for me:

```
Column names in each table must be unique. Column name 'Case_Sens' in table 'sql_server_table' is specified more than once.
```

just FTR I would greatly prefer to not change this.    If the columns are created with case sensitive names then that case sensitivty should be preserved when one queries for those columns in `include_columns`
